config/types/timeouts.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"encoding/json"
+)
+
+const (
+	defaultResponseHeaders int = 10
+	defaultTotal           int = 0
+)
+
+type Timeouts struct {
+	HttpResponseHeaders int `json:"httpResponseHeaders,omitempty"`
+	HttpTotal           int `json:"httpTotal,omitempty"`
+}
+
+type timeouts Timeouts
+
+func (t *Timeouts) UnmarshalJSON(data []byte) error {
+	t.HttpResponseHeaders = defaultResponseHeaders
+	t.HttpTotal = defaultTotal
+	th := timeouts(*t)
```

Comment:
Use a temporary variable for `t` since you are making a copy anyway. No reason to change the Timeouts object.